sootup.core/src/main/java/sootup/core/graph/StmtGraph.java

```diff
@@ -356,6 +520,10 @@ public Iterator<Stmt> iterator() {
   public List<Stmt> getBranchTargetsOf(BranchingStmt fromStmt) {
     final List<Stmt> successors = successors(fromStmt);
     if (fromStmt instanceof JIfStmt) {
+      // Conditional Branch Folder removes neverReachedSucessor Target
+      if (successors.size() == 1) {
+        return Collections.singletonList(successors.get(0));
+      }
```

Comment:
please don't - the state should always be valid and if not you should know you manipulated it.